### PR TITLE
fix: Correctly create directories for paths with forward slashes on Windows

### PR DESCRIPTION
**Pull request by Mixaill**
Fixes sentry-native initialization on Windows when databasepath has forward slashes.

### CODE REVIEW

File: src/path/sentry_path_windows.c [hunk 1]
```diff
@@ -363,10 +363,11 @@ sentry__path_create_dir_all(const sentry_path_t *path)
     memcpy(p, path->path, len * sizeof(wchar_t));
 
     for (ptr = p; *ptr; ptr++) {
-        if (*ptr == '\\' && ptr != p) {
+        if ((*ptr == L'\\' || *ptr == L'/') && ptr != p) {
+            wchar_t temp_char = *ptr;
             *ptr = 0;
             _TRY_MAKE_DIR;
-            *ptr = '\\';
+            *ptr = temp_char;
```
Comment (Swatinem) [inline src/path/sentry_path_windows.c:370]: Might as well just leave it as `\\`

**Comment by Mixaill** [comment]
@Swatinem updated

**Comment by Swatinem** [comment]
Having this tested would be nice. I think its enough to just add a `/qux` or something here: https://github.com/getsentry/sentry-native/blob/5020ff55ac457b02951dd56909424d0fbcda57d0/tests/unit/test_path.c#L10

**Comment by Mixaill** [comment]
@Swatinem tests added. 

I do not known, if backslashes should be tested on non-windows platforms too?

**Comment by Swatinem** [comment]
Sorry for the back and forth here…

Unices are apparently very forgiving when it comes to paths, but you do have an error in there.

```
﻿﻿﻿/home/vsts/work/1/s/tests/unit/test_path.c:150:46: warning: incompatible pointer types passing 'int [4]' to parameter of type 'sentry_pathchar_t *' (aka 'char *') [-Wincompatible-pointer-types]
    sentry_path_t *path_1 = sentry__path_new(L"foo");
                                             ^~~~~~
/home/vsts/work/1/s/src/sentry_path.h:220:37: note: passing argument to parameter 's' here
sentry__path_new(sentry_pathchar_t *s)
                                    ^
```

`sentry__path_new` uses a different typedef for the chars per platform. I think a better test is to compare the behavior of `sentry__path_from_str` with calling `sentry__path_join_str` directly, and see that they match up.

**Comment by Mixaill** [comment]
>sentry__path_new uses a different typedef for the chars per platform

fixed

> I think a better test is to compare the behavior of sentry__path_from_str with calling sentry__path_join_str directly, and see that they match up.

My point that `sentry_path_windows.c::sentry__path_create_dir_all` does not call these functions at all. So it maybe worth to add some kind of test, but it is a bit out of scope for this PR.

**Comment by Swatinem** [comment]
The reason I would like to compare to manually calling `sentry__path_join_str` is that I *think* the testcase is buggy on unix right now.

https://github.com/getsentry/sentry-native/blob/aca8890298f7e73f324463b2cf372e0a60bd7c5f/src/path/sentry_path_unix.c#L322

I *think* unixes will just happily create a directory named `"foo\bar"`, which is *different* from `"foo" / "bar"`.

**Comment by Swatinem** [comment]
Which reminds me… should we maybe normalize separators on unix as well? How common is it to use `\` as separator on unix? Maybe @mitsuhiko can comment here, since you know a lot about path encodings.

**Comment by Mixaill** [comment]
>The reason I would like to compare to manually calling sentry__path_join_str is that I think the testcase is buggy on unix right now.

Thanks, now I understood it. Yeah, it is definitely broken, because `\` is allowed character on unix.

**Comment by mitsuhiko** [comment]
We should only execute tests with `\\` as separators on windows and skip this on Linux. We should not convert backslashes to forward slashes on UNIX. That's unexpected and may cause issues if someone actually wants to use them.

**Comment by Mixaill** [comment]
>We should only execute tests with \\ as separators on windows and skip this on Linux.
Done

**Pull request merged**